### PR TITLE
Automations permissions

### PR DESCRIPTION
**Pull request by znicholasbrown**
PR Checklist:

- [ ] add a short description of what's changed to the top of the `CHANGELOG.md`
- [ ] add/update tests (or don't, for reasons explained below)

## Describe this PR
Adds the proper permissions checks to individual automation features (like system actions and agent SLAs).

### CODE REVIEW

**Comment by whimsicallyson** [comment]
currently reading through the code, any chance we can get a preview link? 😇

**Review by whimsicallyson — CHANGES_REQUESTED** [review]
I'm able to access everything with an enterprise account, and only flow run automations with a starter account. I'm having an issue saving my automation, though (starter account,  save button isn't active):
Screen Shot 2021-04-08 at 10 34 38 AM

also, whoever at vuetify decided to name a class "text-none" instead of "text-transform-none" should get a stern talking-to.

**Comment by madkinsz** [comment]
re @whimsicallyson's screenshot -- `cancel that run` should not be available in starter and should be in the system actions

File: src/utils/automations.js [hunk 1]
```diff
@@ -88,14 +88,20 @@ export const actionTypes = [
 export const featureFlaggedEventTypes = [...actionTypes]
 
 export const flowEventTypes = [
-  { name: 'does not finish', enum: 'STARTED_NOT_FINISHED' },
   {
-    name: 'does not start',
-    enum: 'SCHEDULED_NOT_STARTED'
+    name: 'changes state',
+    enum: 'CHANGES_STATE',
+    permission: 'feature:hooks'
```
Comment (madkinsz) [inline src/utils/automations.js:94]: This feature gate no longer exists, any account can create a state change hook.

File: src/pages/Dashboard/Automations/AddAutomationCard.vue [hunk 2]
```diff
@@ -29,6 +33,10 @@ export default {
         openDuration: { name: 'openDuration', complete: false },
         selectDoThis: { name: 'selectDoThis', complete: false }
       },
+      hookTypes: [
+        { type: 'flow', label: 'flow run', permission: 'feature:hooks' },
```
Comment (madkinsz) [inline src/pages/Dashboard/Automations/AddAutomationCard.vue:37]: `feature:hooks` dropped in https://github.com/PrefectHQ/cloud/pull/3302

Comment (znicholasbrown) [reply src/pages/Dashboard/Automations/AddAutomationCard.vue:37]: Ah yes, this has been updated already in #751 👍 , maybe i should close this PR in favor of that one

Comment (madkinsz) [reply src/pages/Dashboard/Automations/AddAutomationCard.vue:37]: Mm yeah that looks good on that PR 🤷

**Comment by znicholasbrown** [comment]
Closing in favor of #751 - @whimsicallyson I'll look into the issue you mentioned in case it's not already been resolved (Starter tier doesn't have the ability to call the cancel flow run action so that's probably it)

**Comment by whimsicallyson** [comment]
@znicholasbrown I was seeing that issue with either cancel flow run or an email hook, so it might not be?

**Comment by znicholasbrown** [comment]
Fix for the non-disabled cancel run action has been pushed @whimsicallyson - it should be live 👍

**Pull request closed without merging**